backend/src/main/java/ca/bc/gov/hlth/mohums/util/FilteredResponseEntities.java

```diff
@@ -0,0 +1,89 @@
+package ca.bc.gov.hlth.mohums.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * 
+ */
+public class FilteredResponseEntities {
+    
+    private static final HttpStatus OK_STATUS_CODE = HttpStatus.OK;
+
+    private final HttpStatus sourceStatusCode;
+    private final Collection<Object> entities;
+    
+    public static FilteredResponseEntities of(final ResponseEntity<?> responseEntity) {
+        final HttpStatus statusCode = responseEntity.getStatusCode();
+        final FilteredResponseEntities objectResponseEntities;
+
+        if (statusCode.isError()) {
+            objectResponseEntities = new FilteredResponseEntities(statusCode);
+        }
+        else {
+            objectResponseEntities = new FilteredResponseEntities(responseEntity.getBody());
+        }
+        
+        return objectResponseEntities;
+    }
+    
+    FilteredResponseEntities(final HttpStatus statusCode) {
+        this.sourceStatusCode = statusCode;
+        this.entities = Collections.emptyList();
+    }
+    
+    FilteredResponseEntities(final Object responseBody) {
+        this.sourceStatusCode = OK_STATUS_CODE;
+        this.entities = extractEntities(responseBody);
+    }
+    
+    private static Collection<Object> extractEntities(final Object responseBody) {
+        final Collection<Object> entities;
+        
+        if (Objects.isNull(responseBody)) {
+            entities = Collections.emptyList();
+        }
+        else if (responseBody instanceof Collection) {
+            final Collection<?> collection = (Collection) responseBody;
+            entities = collection.stream().filter(Objects::nonNull).collect(Collectors.toList());
+        }
+        else {
+            entities = Collections.singleton(responseBody);
+        }
+        
+        return entities;
+    }
```

Comment:
I suspect this class can be removed or simplified. I'll try to explain in a comment, but I can also screen-share on Monday.

Basically instead of `extractEntities` you can let Spring do the conversion for you. In `WebClientService` we have this method:

```java
private ResponseEntity<Object> get(String path, MultiValueMap<String, String> queryParams) {
    return kcAuthorizedWebClient
            .get()
            .uri(t -> t
                    .path(path)
                    .queryParams(queryParams)
                    .build())
            .exchange().block().toEntity(Object.class).block();
}
```

This works for most of our use cases, but it's inconvenient here where we really want a `List`. Spring can handle that though! Add a new method:

```java
private ResponseEntity<List<Object>> getList(String path, MultiValueMap<String, String> queryParams) {
    return kcAuthorizedWebClient
            .get()
            .uri(t -> t
                    .path(path)
                    .queryParams(queryParams)
                    .build())
            // Here we use toEntityList
            .exchange().block().toEntityList(Object.class).block();
}
```

Use this `getList` method in `getUsers` instead of `get`. Then in `UsersController` you can call:

```java
if (org.isPresent()) {
    List<Object> filteredUsers = users.getBody().stream().filter(new FilterUserByOrgId(org.get())).collect(Collectors.toList());
    return ResponseEntity.status(users.getStatusCode()).body(filteredUsers);
}
```

Because we don't need to change the status code for an empty list, I don't think there is much need for extra handling and the entire `FilteredResponseEntities` can be deleted.

I tested this locally and it seems to work, and the test cases still pass.